src/app/context/wallet/WalletConnectionProvider.tsx

```diff
@@ -1,119 +1,51 @@
-  const { resolvedTheme } = useTheme();
+import { ErrorState } from "@/app/types/errors";
+import { keplrRegistry } from "@/config/wallet/babylon";
 
-  const { mempoolApiUrl, network, networkName } = getNetworkConfig();
-  const bitcoinChain = {
-    id: 1,
-    name: networkName,
-    type: "bitcoin" as any,
-    network: network,
-    backendUrls: {
-      mempoolUrl: mempoolApiUrl + "/api/",
+import { useError } from "../Error/ErrorContext";
+
+const context = typeof window !== "undefined" ? window : {};
+const config = [
+  {
+    chain: "BTC",
+    config: {
+      coinName: "Signet BTC",
+      coinSymbol: "sBTC",
+      networkName: "BTC signet",
+      mempoolApiUrl: "https://mempool.space/signet",
+      network: Network.SIGNET,
     },
-  };
-
-  return (
-    <TomoContextProvider
-      bitcoinChains={[bitcoinChain]}
-      chainTypes={["bitcoin", "cosmos"]}
-      cosmosChains={[
-        {
-          id: 2,
-          name: "Babylon Devnet 7",
-          type: "cosmos",
-          network: "devnet-7",
-          modularData: keplrRegistry,
-          backendUrls: {
-            rpcRrl: "https://rpc.devnet.babylonlabs.io",
-          },
-          logo: "https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/main/images/bbn-dev/chain.png",
-        },
-      ]}
-      style={{
-        rounded: "medium",
-        theme: resolvedTheme as "dark" | "light",
-        primaryColor: "#FF7C2A",
-      }}
-    >
-      <WalletConnectionProviderInternal>
-        {children}
-      </WalletConnectionProviderInternal>
-    </TomoContextProvider>
-  );
-};
-
-interface WalletConnectionContextProps {
-  open: () => Promise<void>;
-  disconnect: () => Promise<void>;
-  isConnected: boolean;
-  providers: ReturnType<typeof useTomoProviders>;
-  tomoWalletConnect: ReturnType<typeof useTomoWalletConnect>;
-  tomoModal: ReturnType<typeof useTomoModalControl>;
-  tomoWalletState: ReturnType<typeof useTomoWalletState>;
-}
-
-const WalletConnectionContext = createContext<WalletConnectionContextProps>({
-  open: async () => {},
-  disconnect: async () => {},
-  isConnected: false,
-  providers: {} as ReturnType<typeof useTomoProviders>,
-  tomoWalletConnect: {} as ReturnType<typeof useTomoWalletConnect>,
-  tomoModal: {} as ReturnType<typeof useTomoModalControl>,
-  tomoWalletState: {} as ReturnType<typeof useTomoWalletState>,
-});
-
-const WalletConnectionProviderInternal = ({ children }: PropsWithChildren) => {
-  const tomoModal = useTomoModalControl();
-  const tomoWalletConnect = useTomoWalletConnect();
-  const tomoWalletState = useTomoWalletState();
-  const providers = useTomoProviders();
-
-  const open = useCallback(async () => {
-    await tomoModal.open("connect");
-  }, [tomoModal]);
-
-  const disconnect = useCallback(async () => {
-    await tomoWalletConnect.disconnect();
-  }, [tomoWalletConnect]);
+  },
+  {
+    chain: "BBN",
+    config: {
+      chainId: keplrRegistry.chainId,
```

Comment:
@jrwbabylonlab this is where we describe the stuff related to the chain
@totraev let's rename `keplrRegistry` to `bbnDevnet`